homeassistant/components/aws/image_processing.py

```diff
@@ -0,0 +1,483 @@
+        return self._confidence
+
+    @property
+    def camera_entity(self):
+        """Return camera entity id from process pictures."""
+        return self.camera
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    async def async_process_objects(self, objects, labels):
+        """Send events with detected objects/labels and store data."""
+        # Send events
+        for target in objects:
+            target_event_data = target.copy()
+            target_event_data[ATTR_ENTITY_ID] = self.entity_id
+            self.hass.bus.fire(EVENT_OBJECT_DETECTED, target_event_data)
+        for label in labels:
+            label_event_data = label.copy()
+            label_event_data[ATTR_ENTITY_ID] = self.entity_id
+            self.hass.bus.fire(EVENT_LABEL_DETECTED, label_event_data)
+        # Update entity store
+        self._objects = objects
+        self._labels = labels
+
+    async def compute_objects(self, client, image):
+        """Detect labels in image and parse response."""
+        detect_labels = await client.detect_labels(Image={"Bytes": image})
+        objects, labels = await get_objects(detect_labels, self.confidence)
+        return objects, labels
+
+    async def async_process_image(self, image):
+        """Process image.
+
+        This method is a coroutine.
+        """
+        async with self.session.create_client(
+            self._service, **self.session_config
+        ) as client:
+            objects, labels = await self.compute_objects(client, image)
+            if len(objects) > 0 and self.conf_save_file_folder:
+                await save_image(
+                    self,
+                    image,
+                    self.name,
+                    objects,
+                    self.conf_save_file_folder,
+                    self.conf_save_file_timestamp,
+                )
+            await self.async_process_objects(objects, labels)
+
+
+class RekognitionFaceEntity(ImageProcessingFaceEntity):
+    """Rekognition Face Detection/Identification capabilities."""
+
+    _service = "rekognition"
+
+    def __init__(
+        self, camera_entity, session, session_config, entity_config, name=None,
+    ):
+        """Initialize the Rekognition service."""
+        super().__init__()
+        self.session = session
+        self.session_config = session_config
+        self.camera = camera_entity
+        self.collection_id = entity_config[CONF_COLLECTION_ID]
+        self.identify_faces = entity_config.get(CONF_IDENTIFY_FACES, False)
+        self.conf_save_file_folder = entity_config.get(CONF_SAVE_FILE_FOLDER, None)
+        self.conf_save_file_timestamp = entity_config.get(
+            CONF_SAVE_FILE_TIMESTAMP, False
+        )
+        self._confidence = entity_config.get(CONF_CONFIDENCE, 0.0)
```

Comment:
from image_processing we have `DEFAULT_CONFIDENCE = 80`

Reply:
Given this is enforced by config validation I've removed the default and just treat it as a required variable